zio-cli/shared/src/main/scala/zio/cli/Options.scala

```diff
@@ -485,6 +486,21 @@ object Options {
         .map(input => self.uid.getOrElse("") :: input.split(" ").toList)
   }
 
+  final case class OAuth2Options(
+    provider: OAuth2Provider,
+    scope: List[String],
+    auxiliaryOptions: Options[OAuth2AuxiliaryOptions]
+  ) extends Options[OAuth2Token] {
+    override def helpDoc: HelpDoc                = auxiliaryOptions.helpDoc
```

Comment:
Rather than delegating everything to `auxiliaryOptions`, it may make sense to provide a bit of help here. But I suppose that depends on how it's rendered and if there is a good place to put help for a phantom options.